src/Init/Data/List/Lemmas.lean

```diff
@@ -2554,4 +2554,17 @@ theorem all_eq_not_any_not (l : List α) (p : α → Bool) : l.all p = !l.any (!
     (l.insert a).all f = (f a && l.all f) := by
   simp [all_eq]
 
+def foldlRecOn {C : β → Sort _} (l : List α) (op : β → α → β) (b : β) (hb : C b)
```

Comment:
```suggestion
@[elab_as_elim]
def foldlRecOn {C : β → Sort _} (l : List α) (op : β → α → β) (b : β) (hb : C b)
```
perhaps `C` should also be renamed to `motive`

Reply:
Err, doesn't work.